src/chain/mod.rs

```diff
@@ -201,9 +252,144 @@ impl ChainSource {
 					}
 				}
 			},
+			Self::BitcoindRpc {
+				bitcoind_rpc_client,
+				header_cache,
+				latest_chain_tip,
+				onchain_wallet,
+				wallet_polling_status,
+				kv_store,
+				config,
+				logger,
+				node_metrics,
+				..
+			} => {
+				// First register for the wallet polling status to make sure `Node::sync_wallets` calls
+				// wait on the result before proceeding.
+				{
+					let mut status_lock = wallet_polling_status.lock().unwrap();
+					if status_lock.register_or_subscribe_pending_sync().is_some() {
+						debug_assert!(false, "Sync already in progress. This should never happen.");
+					}
+				}
+
+				let channel_manager_best_block_hash =
+					channel_manager.current_best_block().block_hash;
+				let sweeper_best_block_hash = output_sweeper.current_best_block().block_hash;
+				let onchain_wallet_best_block_hash = onchain_wallet.current_best_block().block_hash;
+
+				let mut chain_listeners = vec![
+					(
+						onchain_wallet_best_block_hash,
+						&**onchain_wallet as &(dyn Listen + Send + Sync),
+					),
+					(
+						channel_manager_best_block_hash,
+						&*channel_manager as &(dyn Listen + Send + Sync),
+					),
+					(sweeper_best_block_hash, &*output_sweeper as &(dyn Listen + Send + Sync)),
+				];
+
+				// TODO: Eventually we might want to see if we can synchronize `ChannelMonitor`s
```

Comment:
Apologies for the question post-merge but what would be the benefit of syncing the `ChannelMonitor`s before passing them to the `ChainMonitor`? There would still be a need to sync them via the monitor even if it was possible to sync before, right?